packages/compass-preferences-model/src/preferences-schema.ts

```diff
@@ -760,6 +780,52 @@ export const storedUserPreferencesProps: Required<{
     type: 'boolean',
   },
 
+  proxy: {
+    ui: true,
+    cli: true,
+    global: true,
+    description: {
+      short: 'Specify a proxy for Compass to use',
+      long: 'Specify a HTTP, HTTPS or Socks5 proxy to use for connecting to external services (default is picking proxies from environment variables)',
+    },
+    // Internally either a URL or a DevtoolsProxyOptions object as JSON
+    validator: z
+      .union([
+        z.string().url(),
+        z.literal(''),
+        z.custom<string>((val) => {
+          if (typeof val !== 'string') return false;
+          try {
+            JSON.parse(val);
+            return true;
+          } catch {
+            return false;
+          }
+        }),
+      ])
+      .default(''),
+    type: 'string',
+    secrets: {
+      extract(original) {
+        const { secrets, ...remainder } = extractProxySecrets(
+          proxyPreferenceToProxyOptions(original)
+        );
+        return {
+          remainder: JSON.stringify(remainder),
+          secrets: JSON.stringify(secrets),
+        };
+      },
+      merge({ remainder, secrets }) {
+        return proxyOptionsToProxyPreference(
+          mergeProxySecrets({
+            ...JSON.parse(remainder),
+            secrets: JSON.parse(secrets),
+          })
+        );
+      },
+    },
+  },
```

Comment:
This is one of the main two things that I'd like feedback on – how do we feel about this approach to separating secrets from non-secrets in preferences? This is unfortunately not something I considered a lot in the tech design, and I did try another approach first (splitting these into separate preference entries and doing the encryption/decryption in the code that accesses these), but so far it's something I think works pretty well